### PR TITLE
[CI] enable files sharing tests for ocis

### PR DESCRIPTION
**Pull request by dpakach**
## Description

Enable sharing tests for `webUISharingInternalUsers` and `webUISharingPermissionsUsers` suites in CI.

## Related Issue

- Related https://github.com/owncloud/ocis/issues/518

## Motivation and Context

More tests=less bug

## How Has This Been Tested?

:robot: 

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Technical debt
- [x] Tests

## Checklist:

- [ ] Code changes
- [ ] Unit tests added
- [ ] Acceptance tests added
- [ ] Documentation ticket raised: <link> 

## Open tasks:

- [ ] ...

### CODE REVIEW

File: tests/acceptance/helpers/userSettings.js [hunk 1]
```diff
@@ -176,6 +176,14 @@ module.exports = {
...
     }
   },
+
+  getUsernameFromDisplayname: function(username) {
+    for (const userid in this.createdUsers) {
+      if (this.createdUsers[userid].displayname === username) {
+        return userid
+      }
+    }
+  },
```
Comment (phil-davis) [inline tests/acceptance/helpers/userSettings.js:186]:
```suggestion
  getUsernameFromDisplayname: function(displayname) {
    for (const userid in this.createdUsers) {
      if (this.createdUsers[userid].displayname === displayname) {
        return userid
      }
    }
  },
```

Naming is hard. The function is called `getUsernameFromDisplayname` but it takes a parameter `username` and at the end does `return userid` - actually the logic is fine, the names just tricked me for a moment.

File: tests/acceptance/helpers/sharingHelper.js [hunk 2]
```diff
@@ -342,13 +343,39 @@ module.exports = {
...
+    const apiURL = `apps/files_sharing/api/v1/shares/${shareID}`
+    return httpHelper
+      .deleteOCS(apiURL, user)
+      .then(res => {
+        res = httpHelper.checkStatus(res, 'The response status is not the expected value')
+        console.log('deleted Share')
+        return res.json()
+      })
+      .then(res => {
+        return httpHelper.checkOCSStatus(res, 'Could not perform the accept action')
```
Comment (phil-davis) [inline tests/acceptance/helpers/sharingHelper.js:376]:
```suggestion
        return httpHelper.checkOCSStatus(res, 'Could not perform the delete share action')
```

File: tests/acceptance/helpers/sharingHelper.js [hunk 3]
```diff
@@ -342,13 +343,39 @@ module.exports = {
...
           return res.json()
         })
         .then(res => {
+          if (client.globals.ocis) {
+            if (res !== '') {
+              throw new Error(`
+              This is a good error,
+              seems like a bug in ocis has been fixed,
+              just fire up your text editor and remove this line,
+              dont forget to keep your fingers corssed in the meantime.
```
Comment (phil-davis) [inline tests/acceptance/helpers/sharingHelper.js:356]:
```suggestion
              dont forget to keep your fingers crossed in the meantime.
```

File: tests/acceptance/helpers/sharingHelper.js [hunk 4]
```diff
@@ -342,13 +343,39 @@ module.exports = {
...
         .then(res => {
           res = httpHelper.checkStatus(res, 'The response status is not the expected value')
+          if (client.globals.ocis) return res.text()
           return res.json()
         })
         .then(res => {
+          if (client.globals.ocis) {
+            if (res !== '') {
+              throw new Error(`
+              This is a good error,
```
Comment (phil-davis) [inline tests/acceptance/helpers/sharingHelper.js:353]: @dpakach can you explain why this might happen?

Comment (dpakach) [reply tests/acceptance/helpers/sharingHelper.js:353]: Accepting a share in ocis doesn't gives any response right now.
When that is fixed and response is not empty then someone will hopefully see this message and can fix it.
More in https://github.com/owncloud/product/issues/207

**Review by phil-davis — APPROVED** [review]

**Pull request merged**